### PR TITLE
[clang][test] Add missing test file to cmake

### PR DESCRIPTION
**Pull request by keith**
Seems like this test was never running with cmake, but is running with
bazel and broke at head.

### CODE REVIEW

File: clang/unittests/Tooling/RecursiveASTVisitorTests/DeductionGuide.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ Bar s(1);
       DeductionGuideVisitor::Lang_CXX2a));
 }
 
-TEST(RecursiveASTVisitor, DeductionGuideImplicitMode) {
+TEST(RecursiveASTVisitor, DISABLED_DeductionGuideImplicitMode) {
   DeductionGuideVisitor Visitor(/*ShouldVisitImplicitCode*/ true);
   Visitor.ExpectMatch("Foo(T) -> Foo<int>", 11, 1);
   Visitor.ExpectMatch("Bar(type-parameter-0-0) -> Foo<int>", 14, 1);
```
Comment (hokein) [inline clang/unittests/Tooling/RecursiveASTVisitorTests/DeductionGuide.cpp:64]: I think the fix should be straightforward, just adjust the content here. I will fix it afterwards.

**Review by hokein — APPROVED** [review]
oops, good catch!

File: clang/unittests/Tooling/RecursiveASTVisitorTests/DeductionGuide.cpp (hunk 1 above)
Comment (keith) [reply clang/unittests/Tooling/RecursiveASTVisitorTests/DeductionGuide.cpp:64]: thanks! yea i just wasn't confident that i'd make the "right" change here if i just updated the expectation

Comment (hokein) [reply clang/unittests/Tooling/RecursiveASTVisitorTests/DeductionGuide.cpp:64]: I just landed https://github.com/llvm/llvm-project/commit/2ef12b55b2a655262b03204581076522261d5eb3, you can remove the `DISABLED_` now.

Comment (keith) [reply clang/unittests/Tooling/RecursiveASTVisitorTests/DeductionGuide.cpp]: done, thanks!

**Pull request merged**